butterknife-compiler/src/main/java/butterknife/compiler/BindingClass.java

```diff
@@ -350,8 +350,12 @@ private MethodSpec createNewBindToTargetMethod() {
     }
 
     if (bindNeedsFinder()) {
-      result.addParameter(FINDER, "finder")
-          .addParameter(Object.class, "source");
+      if (hasMethodBindings()) {
```

Comment:
Can we make this more exact so that it's only `final` when actually needed inside the anonymous class?